compiler-rt/test/hwasan/TestCases/memcmp_test.cpp

```diff
@@ -0,0 +1,20 @@
+// RUN: %clangxx_hwasan -O0 %s -o %t && not %run %t 2>&1 | FileCheck %s
+// RUN: %clangxx_hwasan -O1 %s -o %t && not %run %t 2>&1 | FileCheck %s
+// RUN: %clangxx_hwasan -O2 %s -o %t && not %run %t 2>&1 | FileCheck %s
+// RUN: %clangxx_hwasan -O3 %s -o %t && not %run %t 2>&1 | FileCheck %s
+
+// REQUIRES: compiler-rt-optimized
+
+// FIXME: we need to implement memcmp intercepptor to make this work.
+// UNSUPPORTED: hwasan
+
+#include <string.h>
+int main(int argc, char **argv) {
+  char a1[] = {static_cast<char>(argc), 2, 3, 4};
```

Comment:
if you do heap allocation we can test in aliasing mode

Reply:
// XFAIL: *

I believe "// UNSUPPORTED: hwasan" does not work in hwasan, this is sanitizer_common feature.
I guess check-compiler-rt does not pass with this patch

Reply:
I guess it's worth porting  bcmp_test.cpp and installing bcmp interceptor

memcmp_strict_test - maybe for consistencly

Reply:
builder unhappy https://github.com/llvm/llvm-project/actions/runs/6280283712/job/17057273713

Reply:
I think I will implement the feature and include the test with in instead of disabling the test.

Reply:
bcmp ?